src/main/scala/com/fulcrumgenomics/umi/UmiConsensusCaller.scala

```diff
@@ -241,16 +246,27 @@ trait UmiConsensusCaller[ConsensusRead <: SimpleRead] {
       }
     }
 
-    // Find the last non-N base of sufficient quality in the record, starting from either the
-    // end of the read, or the end of the insert, whichever is shorter!
+    // Get the length of the read based on trimming bases that map beyond the mate's end (FR only) and then any
+    // remaining trailing Ns
     val len = {
       var index = if (!rec.isFrPair) trimToLength - 1 else {
-        // Adjust the insert size based on clipping on the 5' end of the given read.
-        // This *does not* consider any clipping on the 5' end of its mate.
-        val fivePrimeClipping  = cigar.takeWhile(_.operator.isClipping).filter(_.operator == CigarOperator.S).sumBy(_.length)
-        val adjustedInsertSize = abs(rec.insertSize) + fivePrimeClipping
-        min(adjustedInsertSize, trimToLength) - 1
+        // Get the number of mapped bases to clip that maps beyond the mate's end.  Soft-clipped bases are not counted.
+        val numBasesExtendingPasteMateEnd = this.clipper.clipExtendingPastMateEnd(rec=rec, doNotClip=true)
+        if (numBasesExtendingPasteMateEnd == 0) trimToLength - 1
```

Comment:
this matches the previous behavior.  We don't trim if only soft-clipped bases are "beyond the end of the mate"